finesse/hardware/stepper_motor/stepper_motor_base.py

```diff
@@ -95,10 +95,23 @@ def notify_on_stopped(self) -> None:
         The message is stepper.move.end.
         """
 
+    @property
+    @abstractmethod
+    def is_moving(self) -> bool:
+        """Whether the motor is currently moving."""
+
     @property
     def angle(self) -> float:
-        """The current angle of the motor in degrees."""
-        return self.step * 360.0 / self.steps_per_rotation
+        """The current angle of the motor in degrees.
+
+        Returns:
+            The current angle or NaN if the stepper motor is moving
+        """
+        step = self.step
+        if step is None:
+            return float("nan")
```

Comment:
This is done like this so there's something to output to the file while moving, but what's the meaning of outputing nan as an angle? Why not just not outputing anything?

Reply:
You mean return `None`? I did consider doing that, but this was slightly less effort :wink:. The nice thing about returning `None` though is we get the extra type-checking from `mypy`... I'll do that and save the `nan` conversion for when it's being written to the data file.